src/models/nondifferentiable/mlj_tree.jl

```diff
@@ -0,0 +1,235 @@
+using MLJDecisionTreeInterface
+using DataFrames
+using MLJBase
+
+# The implementation of MLJ: DecisionTree: https://github.com/JuliaAI/DecisionTree.jl/blob/dev/src/classification/main.jl
+
+"""
+    TreeModel <: AbstractNonDifferentiableJuliaModel
+
+Constructor for tree-based models from the MLJ library. 
+
+# Arguments
+- `model::Any`: The model selected by the user. Must be a model from the MLJ library.
+- `likelihood::Symbol`: The likelihood of the model. Must be one of `[:classification_binary, :classification_multi]`.
+
+# Returns
+- `TreeModel`: A tree-based model from the MLJ library wrapped inside the TreeModel class.
+"""
+struct TreeModel <: AbstractNonDifferentiableJuliaModel
+    model::Any
+    likelihood::Symbol
+    function TreeModel(model, likelihood)
+        if !(
+            model.model isa MLJDecisionTreeInterface.DecisionTreeClassifier ||
+            model.model isa MLJDecisionTreeInterface.RandomForestClassifier
+        )
+            throw(
```

Comment:
Again some hideous formatting, let me know if this was the auto-formatter or you.

Reply:
This was definitely the auto-formatter, can't fix this I think